homeassistant/components/sensor/logi_circle.py

```diff
@@ -0,0 +1,154 @@
+"""
+This component provides HA sensor support for Logi Circle cameras.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.logi_circle/
+"""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.logi_circle import (
+    CONF_ATTRIBUTION, DEFAULT_ENTITY_NAMESPACE, DOMAIN)
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_ENTITY_NAMESPACE, CONF_MONITORED_CONDITIONS,
+    STATE_UNKNOWN, ATTR_ATTRIBUTION)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.icon import icon_for_battery_level
+from homeassistant.util.dt import as_local
+
+DEPENDENCIES = ['logi_circle']
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+# Sensor types: Name, unit of measure, icon per sensor key.
+SENSOR_TYPES = {
+    'battery_level': [
+        'Battery', '%', 'battery-50'],
+
+    'last_activity_time': [
+        'Last Activity', None, 'history'],
+
+    'privacy_mode': [
+        'Privacy Mode', None, 'eye'],
+
+    'signal_strength_category': [
+        'WiFi Signal Category', None, 'wifi'],
+
+    'signal_strength_percentage': [
+        'WiFi Signal Strength', '%', 'wifi'],
+
+    'speaker_volume': [
+        'Volume', '%', 'volume-high'],
+
+    'streaming_mode': [
+        'Streaming Mode', None, 'camera'],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_ENTITY_NAMESPACE, default=DEFAULT_ENTITY_NAMESPACE):
+        cv.string,
+    vol.Required(CONF_MONITORED_CONDITIONS, default=list(SENSOR_TYPES)):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+})
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up a sensor for a Logi Circle device."""
+    devices = hass.data[DOMAIN]
+
+    sensors = []
+    for sensor_type in config.get(CONF_MONITORED_CONDITIONS):
+        for device in devices:
+            if device.supports_feature(sensor_type):
+                sensors.append(LogiSensor(hass, device, sensor_type))
+
+    async_add_entities(sensors, True)
+
+
+class LogiSensor(Entity):
+    """A sensor implementation for a Logi Circle camera."""
+
+    def __init__(self, hass, camera, sensor_type):
```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant. Pass in the timezone instead.